enhancements/monitoring/monitoring-stack-operator.md

```diff
@@ -0,0 +1,490 @@
+* Allow logging and tracing capabilities. Other solutions should tackle this (designed by Logging and Tracing teams.
+
+## Proposal
+
+### Personas
+
+* OSD/Platform SRE
+* MTSRE
+* RHOBS Devs
+* MTO
+* Customer: User of Manage Service and OSD
+
+### User Stories
+
+#### Story 1
+
+As a MTSRE/MTO, I would like to define ServiceMonitors that will scrape metrics from my managed service by a Prometheus-compatible system.
+
+#### Story 2
+
+As an MTSRE/MTO, I would like to query SaaS/Addon metrics using PromQL to understand the health or troubleshoot the performance of my managed service.
+
+#### Story 3
+
+As an MTSRE/MTO, I would like to create alerting and recording rules for Prometheus metrics exposed by my managed service.
+
+#### Story 4
+
+As an MTSRE, I would like to configure routing for alerts defined on Prometheus metrics exposed by my SaaS/Addon.
+
+#### Story 5
+
+As an MTO, I would like to query certain metrics for my service across all clusters in which the managed service is deployed or used.
+
+#### Story 6
+
+As an MTO, I would like to create dashboards for certain metrics across all clusters where the SaaS/Addon is deployed or used.
+
+#### Story 7
+
+As a Customer, I can use UWM without seeing Addons metrics, alerting or recording configurations.
+
+### Implementation Details/Notes/Constraints
+
+Since UWM and PM are part of Openshift Core, we would like to avoid increasing the complexity and introducing potential bugs for customers by extending those monitoring stacks. Because of this, we propose developing a new stack that will be able to support the managed service use-case while staying compatible with upcoming use-cases such as Hypershift or [KCP](https://github.com/kcp-dev/kcp).
+
+#### Overview
+
+We propose developing a new open-source operator with a working name, “MonitoringStack Operator” (official name proposals are welcome!) that manages a `MonitoringStack` CRD and can be used to deploy an independent, prometheus-based stack for each managed service.
+The operator will be installed into its own namespace and will watch `MonitoringStack` resources in the entire cluster. Alongside itself, the operator will deploy two other operators:
+
+* A Prometheus Operator in order to take advantage of the Prometheus management functionality already present in PO. An Alertmanager instance for managing alert routing will also be provisioned in the same namespace.
+* A [Grafana Operator](https://operatorhub.io/operator/grafana-operator)
+
+Once a `MonitoringStack` has been created, the operator will deploy the appropriate resources in the `MonitoringStackOperor`'s namespace based on how the stack is configured. In a simple case, this might be:
+
+* a Prometheus CR
+* a Grafana CR and a Grafana Data Source CR
+* a Thanos Querier if the stack is configured for high availability
+
+In other cases, it might be just a Prometheus Agent deployment (forwarding only capabilities).
```

Comment:
To me it is unclear which cases would deploying just this agent would take precedent over the former cases, would it be possible to extend this with an example or clarification?

Also is this already an existing solution or is this reference to this work in progress https://github.com/prometheus/prometheus/pull/8785? Maybe adding a note / link could be useful as well.

Reply:
I read this as aligning this design with what we will need in HyperShift, where we don't want or need a full-blown monitoring stack on every cluster.  This way on a traditional topology, the MonitoringStack CR would state deploy the full-blown stack and on a hypershift topology, the MonitoringStack CR would deploy an agent that forwards off-cluster.